src/com/cft/contactmerge/NameMatchSelector.java

```diff
@@ -0,0 +1,75 @@
+/* Modified by James Page on May 8th, 2018
+ * Contains the selections of which type of naming properties will be used for matching
+ */
+
+package com.cft.contactmerge;
+
+import java.util.ArrayList;
+
+public abstract class NameMatchSelector extends NameMatchLogic {
+    /************************************************************************
+     *********** Method(s) used to call specific name comparisons ***********
+     ************************************************************************/
+    public static AnswerType doFirstNamesMatchUsingAllComparisonChecks(String contactOneFirstName, String contactTwoFirstName){
+        // edit names to go through simple matching algorithms
+        ArrayList<String> editedFirstNameOne = editNameForMatching(contactOneFirstName);
+        ArrayList<String> editedFirstNameTwo = editNameForMatching(contactTwoFirstName);
+
+        return FirstNameMatcher.getFirstNameMatchResult(editedFirstNameOne, editedFirstNameTwo);
+    }
+
+    public static AnswerType doLastNamesMatchUsingAllComparisonChecks(String contactOneLastName, String contactTwoLastName){
+        ArrayList<String> editedLastNameOne = editNameForMatching(contactOneLastName);
+        ArrayList<String> editedLastNameTwo = editNameForMatching(contactTwoLastName);
+
+        return LastNameMatcher.getLastNameMatchResult(editedLastNameOne, editedLastNameTwo);
+    }
+
+    public static AnswerType doFirstAndLastNamesMatchUsingAllComparisonChecks(String contactOneFirstName, String contactOneLastName, String contactTwoFirstName, String contactTwoLastName){
+
+        // edit names to go through simple matching algorithms
+        ArrayList<String> editedFirstNameOne = editNameForMatching(contactOneFirstName);
+        ArrayList<String> editedLastNameOne = editNameForMatching(contactOneLastName);
+        ArrayList<String> editedFirstNameTwo = editNameForMatching(contactTwoFirstName);
+        ArrayList<String> editedLastNameTwo = editNameForMatching(contactTwoLastName);
+
```

Comment:
We normalize each name part twice with this logic. Is there a way to do it just once?

Reply:
There is 4 names total:
the "One" represents recordOne, and "Two" represents recordTwo. Both record names have both first and last names. This requires splitting 4 names into parts (as needed):
* contactOneFirstName
* contactOneLastName
* contactTwoFirstName
* contactTwoLastName
... each one is only called once. These four names are only converted into collective parts one time (each), and then passed through the method(s) responsible for finding matches. That being said, I'm all for a better naming method if you can think of any. We'll talk about this more in the meeting tomorrow.